balance.go

```diff
@@ -19,6 +31,7 @@ type Balance struct {
 
 // Amount is a structure wrapping an amount value and its currency.
 type Amount struct {
-	Value    int64    `json:"amount"`
-	Currency Currency `json:"currency"`
+	Value       int64                       `json:"amount"`
+	Currency    Currency                    `json:"currency"`
+	SourceTypes map[BalanceSourceType]int64 `json:"source_types"`
```

Comment:
@brandur-stripe What do you think of doing this versus using a sub-class to represent the mapping?

In Java, we have a class named `Money` for the `Amount` class ([here](https://github.com/stripe/stripe-java/blob/master/src/main/java/com/stripe/model/Balance.java#L104)) and that has a sub-class named `SourceType`([here](https://github.com/stripe/stripe-java/blob/master/src/main/java/com/stripe/model/Balance.java#L122) (which really should be named `SourceType` instead.).

In dotnet we have `BalanceAmount` ([here](https://github.com/stripe/stripe-dotnet/blob/master/src/Stripe.net/Entities/Balance/BalanceAmount.cs)) (which is what `Amount` in Go should be named really if we were consistent). And that class doesn't have a sub-class for `SourceTypes` and instead uses a `Dictionary` like is introduced in this PR

Reply:
@remi-stripe I'd personally say that this approach is okay.

I don't like Java's because it makes accessing information less convenient — you have to search the list instead of just being able to index on a map.

I could see an argument for maybe making the `int64` another substruct in case it ever becomes a subobject in the API, but IMO it'd be okay to leave it like this and just change it later if necessary.